lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery.hs

```diff
@@ -127,15 +127,15 @@ class CompareDiscovery s where
         -> Address
         -> Ordering
 
--- | Extract the list of all known addresses.
+-- | Extract the list of all known addresses. Also specify Role/Index.
 --
 -- NOTE: Change addresses aren't considered "known" until they've been used. The
 -- rationale is that, we don't want users or consumers of the wallet to be using
 -- change addresses prematurely.
 class KnownAddresses s where
     knownAddresses
         :: s
-        -> [(Address, AddressState)]
+        -> [(Address, AddressState, DerivationIndex, DerivationIndex)]
```

Comment:
Shouldn't the result be a `NonEmpty DerivationIndex` like it's done for `IsOurs`? Derivation paths actually have different length depending on the type of wallets. Byron wallets have 2 indexes, Icarus and Shelley have 5.